### DISCUSSION TITLE
Ship of Garbo Paradox

### DISCUSSION THREAD

**Discussion by gomigg07** [TurboWarp/extensions#1041] [General]
How many features can we add to Turbowarp to the point when we can no longer call it a Scratch mod and it becomes it's own engine? Just some food for thought.

**Comment by snowboyz0825** [comment]
At this point, I don't even consider it a mod. I can't live without the custom stage size, Lily's toolbox, comments, and tons of other things. Turbowarp is most definitely it's own thing independent of scratch at this point, it's just scratch based, like how scratch is javascript based.

**Reply by CST1229** [reply]
> Lily's toolbox

obligatory "don't use lms utilities/lily's toolbox, use other extensions, they're better"

**Reply by snowboyz0825** [reply]
It's mostly just that I know what everything is and I can't imagine using anything else at this point, plus it has stuff from a lot of other extensions that I don't wanna get individually and have to remember what they are and where to find them, etc

**Comment by Drago-Cuven** [comment]
i gotta agree. turbowarp is its own thing by now. At this point just drop
scratch compatibility and turn this into a game engine.

**Comment by penta-quark-neutro** [comment]
I always saw turbowarp as a game engine and not as a scratch mod

**Comment by NexusKitten** [comment]
> How many features can we add to Turbowarp to the point when we can no longer call it a Scratch mod and it becomes it's own engine? Just some food for thought.

I'm not sure exactly what's Garbo's outlook on it at this point, however considering we're getting more features like custom reporters, custom fonts, etc, it feels like TurboWarp is more of it's own thing that just happens to be backwards compatible with Scratch. Scratch itself had so much potential, but the Scratch Team decided to market it as a learning platform and nothing more-- I see TurboWarp as an experiment of what happens when that silly mindset is removed.

**Reply by gomigg07** [reply]
This is a good way to look at it!

**Comment by yuri-kiss** [comment]
i do agree, i look at it the same way, it would seem most people who use it often would most likely agree, but yeah i would like to see the "compatibility" factor be lessened seeing as the stuff we have now is not really "compatible".
________________________________

**Reply by gomigg07** [reply]
I think at this point we could just make a seperate experimental fork that drops all Scratch compatibility and goes ham on all the best features, especially with the costume editor and handling sprite animations. I'd love to make it myself but I only have the most basic coding knowledge

**Comment by yuri-kiss** [comment]
many people have had this idea, i support it, i mean garbo owns unsandboxed.org(i think its .org)
________________________________

**Comment by BlueDome77** [comment]
And to think that TurboWarp is still compatible with Scratch projects! (aside from custom extensions)

**Comment by PPPDUD** [comment]
Hot take: TurboWarp is still a Scratch mod, just a very customizable one.
